cmd/argo/commands/resubmit.go

```diff
@@ -39,31 +71,77 @@ func NewResubmitCommand() *cobra.Command {
-
-			for _, name := range args {
-				created, err := serviceClient.ResubmitWorkflow(ctx, &workflowpkg.WorkflowResubmitRequest{
-					Namespace: namespace,
-					Name:      name,
-					Memoized:  memoized,
-				})
-				errors.CheckError(err)
-				printWorkflow(created, getFlags{output: cliSubmitOpts.output})
-				waitWatchOrLog(ctx, serviceClient, namespace, []string{created.Name}, cliSubmitOpts)
-			}
+			resubmitOpts.namespace = client.Namespace()
+			err := resubmitWorkflows(ctx, serviceClient, resubmitOpts, cliSubmitOpts, args)
+			errors.CheckError(err)
 		},
 	}
 
-	command.Flags().Int32Var(&priority, "priority", 0, "workflow priority")
+	command.Flags().Int32Var(&resubmitOpts.priority, "priority", 0, "workflow priority")
 	command.Flags().StringVarP(&cliSubmitOpts.output, "output", "o", "", "Output format. One of: name|json|yaml|wide")
-	command.Flags().BoolVarP(&cliSubmitOpts.wait, "wait", "w", false, "wait for the workflow to complete")
-	command.Flags().BoolVar(&cliSubmitOpts.watch, "watch", false, "watch the workflow until it completes")
+	command.Flags().BoolVarP(&cliSubmitOpts.wait, "wait", "w", false, "wait for the workflow to complete, only works when a single workflow is resubmitted")
+	command.Flags().BoolVar(&cliSubmitOpts.watch, "watch", false, "watch the workflow until it completes, only works when a single workflow is resubmitted")
 	command.Flags().BoolVar(&cliSubmitOpts.log, "log", false, "log the workflow until it completes")
-	command.Flags().BoolVar(&memoized, "memoized", false, "re-use successful steps & outputs from the previous run (experimental)")
+	command.Flags().BoolVar(&resubmitOpts.memoized, "memoized", false, "re-use successful steps & outputs from the previous run (experimental)")
+	command.Flags().StringVarP(&resubmitOpts.labelSelector, "selector", "l", "", "Selector (label query) to filter on, not including uninitialized ones, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	command.Flags().StringVar(&resubmitOpts.fieldSelector, "field-selector", "", "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
 	return command
 }
+
+// resubmitWorkflows resubmits workflows by given resubmitOpts or workflow names
+func resubmitWorkflows(ctx context.Context, serviceClient workflowpkg.WorkflowServiceClient, resubmitOpts resubmitOps, cliSubmitOpts cliSubmitOpts, args []string) error {
+	var (
+		wfs wfv1.Workflows
+		err error
+	)
+	if resubmitOpts.hasSelector() {
+		wfs, err = listWorkflows(ctx, serviceClient, listFlags{
+			namespace: resubmitOpts.namespace,
+			fields:    resubmitOpts.fieldSelector,
+			labels:    resubmitOpts.labelSelector,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, n := range args {
+		wfs = append(wfs, wfv1.Workflow{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      n,
+				Namespace: resubmitOpts.namespace,
+			},
+		})
+	}
```

Comment:
Some of the logic for the CLI selectors can definitely be reused.

I should create a follow-up PR to have the common logic in `common.go` and use it across all commands that support selectors.